File: src/main/java/app/askresume/domain/resume/validator/ResumeValidator.java
package app.askresume.domain.resume.validator;

import app.askresume.domain.resume.constant.DifficultyType;
import app.askresume.global.error.ErrorCode;
import app.askresume.global.error.exception.BusinessException;
import org.springframework.stereotype.Service;

@Service
public class ResumeValidator {
    public void validateDifficultyType(String type) {
        if (!DifficultyType.isDifficultyType(type)) {
            throw new BusinessException(ErrorCode.INVALID_DIFFICULTY_TYPE);
        }
    }
}

File: src/main/kotlin/app/askresume/domain/resume/validator/ResumeValidator.kt
package app.askresume.domain.resume.validator

import app.askresume.domain.resume.constant.DifficultyType
import app.askresume.global.error.ErrorCode
import app.askresume.global.error.exception.BusinessException
import org.springframework.stereotype.Service

@Service
class ResumeValidator {
    fun validateDifficultyType(type: String) {
        if (!DifficultyType.isDifficultyType(type)) {
            throw BusinessException(ErrorCode.INVALID_DIFFICULTY_TYPE)
        }
    }
}
